### ISSUE TITLE
enable istio on the tenant components

### ISSUE THREAD

**Issue by sjmiller609** [astronomer/terraform-google-astronomer-cloud#21]
Istio is enabled on the platform components, but not on the multi-tenant components. Let's enable it on everything.

**Comment by sjmiller609** [comment]
enabling it on the ingress nginx worked fine, but it did not work with fluentd out of the box, regardless of gvisor. https://github.com/sjmiller609/helm.astronomer.io/commit/8d7e8b53d9d402eb8297cd710949bec1559f013e

There is this doc about fluentd with istio, which will require some investigation, and I will do this components last https://istio.io/docs/tasks/telemetry/logs/fluentd/

**Comment by sjmiller609** [comment]
Allow commander to set labels on a namespace https://github.com/astronomer/commander/pull/13

**Comment by sjmiller609** [comment]
Use cluster networking to connect from houston to the registry https://github.com/astronomer/houston-api/pull/108

**Comment by sjmiller609** [comment]
Send airflow namespace labels to commander from houston https://github.com/astronomer/houston-api/issues/103

**Comment by sjmiller609** [comment]
Running with istio locally using the above 3 PRs. There is an issue where the istio sidecar containers are below the minimum mem request for a container in the namespace because of limit ranges.
```
Stevens-MacBook-Pro:helm.astronomer.io steven$ kubectl describe replicaset -n astronomer-weightless-perturbation-6979 weightless-perturbation-6979-webserver-6c7dbdfb7b
Name:           weightless-perturbation-6979-webserver-6c7dbdfb7b
Namespace:      astronomer-weightless-perturbation-6979
...
  Warning  FailedCreate  6m53s (x9 over 17m)  replicaset-controller  (combined from similar events): Error creating: pods "weightless-perturbation-6979-webserver-6c7dbdfb7b-94xsp" is forbidden: [minimum memory usage per Container is 384Mi, but request is 128Mi., minimum cpu usage per Container is 100m, but request is 10m., minimum memory usage per Container is 384Mi, but request is 10Mi.]

Stevens-MacBook-Pro:helm.astronomer.io steven$ kubectl describe limitrange -n astronomer-weightless-perturbation-6979 weightless-perturbation-6979-limit-range
Name:       weightless-perturbation-6979-limit-range
Namespace:  astronomer-weightless-perturbation-6979
Type        Resource  Min    Max      Default Request  Default Limit  Max Limit/Request Ratio
----        --------  ---    ---      ---------------  -------------  -----------------------
Pod         cpu       -      3500m    -                -              -
Pod         memory    -      13440Mi  -                -              -
Container   cpu       100m   -        100m             100m           -
Container   memory    384Mi  -        384Mi            384Mi          -
```

**Comment by sjmiller609** [comment]
I think the above problem is related to billing and in houston
```
function limitRange() {
```

**Comment by sjmiller609** [comment]
I first need to understand why there is a container limit range on the low side

**Comment by sjmiller609** [comment]
On slack there was a discussion and we don't exactly know if there is a requirements for a minimum, but we do want a default of 1 AU worth of resources. Now I am running into a different issue related to resource quotas. I need to come to understand the difference between a limit range and a resource quota
```
Error creating: pods "weightless-perturbation-6979-webserver-6c7dbdfb7b-qknpf" is forbidden: exceeded quota: weightless-perturbation-6979-resource-quota, requested: limits.cpu=2500m, used: limits.cpu=2300m, limited: limits.cpu=4200m
Stevens-MacBook-Pro:helm.astronomer.io steven$ kubectl describe limitranges -n astronomer-weightless-perturbation-6979 weightless-perturbation-6979-limit-range
```

**Comment by sjmiller609** [comment]
I think resource quota applies to the whole namespace, and that's the primary difference

**Comment by sjmiller609** [comment]
the issue is that the default "limit" on istio sidecar containers is
```
      limits:
        cpu: "2"
        memory: 1Gi
```
but they likely require far less considering that the 'requests' are
```
      requests:
        cpu: 100m
        memory: 128Mi
```

**Comment by sjmiller609** [comment]
I may reconfigure the defaults for the side limits using helm values https://github.com/istio/istio/blob/master/install/kubernetes/helm/istio/values.yaml Istio docs https://istio.io/docs/concepts/performance-and-scalability/ state that "The Envoy proxy uses 0.6 vCPU and 50 MB memory per 1000 requests per second going through the proxy."

**Comment by sjmiller609** [comment]
One issue is that since different components will have different amounts of traffic, we would prefer a way to configure istio sidecar/proxy per pod rather than for all sidecars. There is a way to do it for requests https://github.com/istio/istio/issues/16126

**Comment by sjmiller609** [comment]
One potentially relevant option is to use LimitRangeItem.maxLimitRequestRatio - max limit can be set relative to the request, but I don't think this will provide a default value on the pod, only restrict pods in violation from being placed.

**Comment by sjmiller609** [comment]
a bit more information about the memory consumption - looks like memory should grow when the total number of reachable services from that pod increases, not when traffic increases.

**Comment by schnie** [comment]
@sjmiller609 now that we (almost) have all the PRs merged in, what is left that we know of? Seems like we'll be able to pull together a build of master and try everything out?

**Comment by sjmiller609** [comment]
yep, I'm giving it a shot. there are some things I know will need to be tweaked - specifically how AUs and billing works. Then there are also still some components that do not use istio.
- ingress: just need to decide if we like Istio-style ingress gateways
- fluentd: some issue, not understood yet
- elasticsearch-curator: some issue, not understood yet
